tests/Collection/CollectionFunctionalTest.php

```diff
@@ -180,6 +182,41 @@ public function testWithOptionsPassesOptions()
         $this->assertSame(WriteConcern::MAJORITY, $debug['writeConcern']->getW());
     }
 
+    public function testMapReduce()
+    {
+        $this->createFixtures(3);
+
+        $map = new Javascript('function() { emit(this._id, this.x); }');
+        $reduce = new Javascript('function(key, values) { return Array.sum(values); }');
+        $out = ['inline' => 1];
+
+        $result = $this->collection->mapReduce($map, $reduce, $out);
+
+        $this->assertInstanceOf('MongoDB\MapReduceResult', $result);
+        $this->assertGreaterThanOrEqual(0, $result->getExecutionTimeMS());
+        $this->assertNotEmpty($result->getCounts());
+        $this->assertNotEmpty($result->getTiming());
+    }
+
+    public function testMapReduceWithOptions() {
+        $this->createFixtures(3);
+
+        $map = new Javascript('function() { emit(this._id, this.x); }');
+        $reduce = new Javascript('function(key, values) { return Array.sum(values); }');
+        $out = ['inline' => 1];
+
+        $options = ["readPreference" => ReadPreference::RP_SECONDARY_PREFERRED];
+
+        $operation = new MapReduce($this->getDatabaseName(),
```

Comment:
This shouldn't be testing the Operation class directly, as the test doesn't interact with the Collection object at all.

I think we can just remove the `testMapReduce()` test, as the read preference option isn't testing much. In most environments, this is just going to select the primary server.